diff --git a/components/wireless/wifi6/wpa_supplicant/CMakeLists.txt b/components/wireless/wifi6/wpa_supplicant/CMakeLists.txt
index f647e0ca..99a6cf32 100644
--- a/components/wireless/wifi6/wpa_supplicant/CMakeLists.txt
+++ b/components/wireless/wifi6/wpa_supplicant/CMakeLists.txt
@@ -16,7 +16,9 @@
 endif()
 
 sdk_library_add_sources(wpa_supplicant/wpa_supplicant.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(wpa_supplicant/wps_supplicant.c)
+endif()
 sdk_library_add_sources(wpa_supplicant/events.c)
 sdk_library_add_sources(wpa_supplicant/scan.c)
 sdk_library_add_sources(wpa_supplicant/notify.c)
@@ -56,12 +58,16 @@
 sdk_library_add_sources(src/wpa_crypto/dh_group5.c)
 #sdk_library_add_sources(src/wpa_crypto/crypto_openssl.c)
 #sdk_library_add_sources(src/wpa_crypto/crypto_libtomcrypt.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/wpa_crypto/crypto_internal-modexp.c)
+endif()
 sdk_library_add_sources(src/tls/bignum.c)
 sdk_library_add_sources(src/wpa_crypto/dh_group5.c)
 #sdk_library_add_sources(src/wpa_crypto/crypto_openssl.c)
 #sdk_library_add_sources(src/wpa_crypto/crypto_libtomcrypt.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/wpa_crypto/crypto_internal-modexp.c)
+endif()
 sdk_library_add_sources(src/tls/bignum.c)
 sdk_library_add_sources(wpa_supplicant/sme.c)
 sdk_library_add_sources(src/common/sae.c)
@@ -127,6 +133,7 @@
 sdk_library_add_sources(src/ap/wnm_ap.c)
 endif()
 sdk_library_add_sources(src/eap_server/eap_server.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/ap/wps_hostapd.c)
 sdk_library_add_sources(src/wps/wps.c)
 sdk_library_add_sources(src/wps/wps_attr_build.c)
@@ -134,16 +141,23 @@
 sdk_library_add_sources(src/wps/wps_attr_parse.c)
 sdk_library_add_sources(src/wps/wps_registrar.c)
 sdk_library_add_sources(src/wps/wps_common.c)
+endif()
 sdk_library_add_sources(src/eap_server/eap_server_identity.c)
 sdk_library_add_sources(src/eap_server/eap_server_methods.c)
 sdk_library_add_sources(src/eap_common/eap_common.c)
 sdk_library_add_sources(src/eap_peer/eap.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/eap_peer/eap_wsc.c)
+endif()
 sdk_library_add_sources(src/eap_peer/eap_methods.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/eap_common/eap_wsc_common.c)
+endif()
 sdk_library_add_sources(src/ap/ctrl_iface_ap.c)
+if(NOT CONFIG_BRIDGE_DISABLE_WPS)
 sdk_library_add_sources(src/wps/wps_attr_process.c)
 sdk_library_add_sources(src/wps/wps_enrollee.c)
+endif()
 
 sdk_add_include_directories(wpa_supplicant)
 sdk_add_include_directories(src)
diff --git a/components/wireless/wifi6/wpa_supplicant/src/utils/build_config.h b/components/wireless/wifi6/wpa_supplicant/src/utils/build_config.h
index b9c5580b..3e2ce49d 100644
--- a/components/wireless/wifi6/wpa_supplicant/src/utils/build_config.h
+++ b/components/wireless/wifi6/wpa_supplicant/src/utils/build_config.h
@@ -107,17 +107,21 @@
 #if MACSW_WFA
 #define CONFIG_WNM_AP
 #endif
+#ifndef CONFIG_BRIDGE_DISABLE_WPS
 #define CONFIG_WPS
 #define EAP_WSC
+#endif
 #if MACSW_P2P
 #define CONFIG_P2P
+#ifndef CONFIG_BRIDGE_DISABLE_WPS
 #define CONFIG_WPS
+#define EAP_WSC
+#define EAP_SERVER_WSC
+#endif
 #define CONFIG_GAS
 #define CONFIG_IEEE8021X_EAPOL
 #define CONFIG_SHA256
 #define LTM_FAST
-#define EAP_WSC
-#define EAP_SERVER_WSC
 #define CONFIG_OFFCHANNEL
 #define IEEE8021X_EAPOL
 #endif // MACSW_P2P
 #if MACSW_VHT
 #define CONFIG_IEEE80211AC
 #endif
