--- a/components/wireless/wifi6/wpa_supplicant/src/drivers/driver_macsw.c
+++ b/components/wireless/wifi6/wpa_supplicant/src/drivers/driver_macsw.c
@@ -1484,6 +1484,8 @@
 	struct mac_chan_def *chan_tag;
 	int i;
 
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(246, 0);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(136, 0);
 	if (feat->chan->chan2G4_cnt > ARRAY_SIZE(macsw_2g_channels))
@@ -1491,6 +1493,8 @@
 	mode->mode = hw_mode;
 	mode->num_channels = feat->chan->chan2G4_cnt;
 	mode->channels = macsw_2g_channels;
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(247, feat->chan->chan2G4_cnt);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G) {
 		fhost_bridge_ap_start_progress(137, feat->chan->chan2G4_cnt);
 		fhost_bridge_ap_start_progress(138, 0);
@@ -1498,18 +1502,26 @@
 
 	chan = mode->channels;
 	chan_tag = feat->chan->chan2G4;
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(248, feat->chan->chan2G4_cnt);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(139, feat->chan->chan2G4_cnt);
 	for (i = 0; i < feat->chan->chan2G4_cnt; i++, chan++, chan_tag++)
 		macsw_to_hostapd_channel(chan_tag, chan);
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(249, i);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(140, i);
 
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(251, is_11b_only_rates);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(141, is_11b_only_rates);
 	mode->rates = macsw_init_rates(&mode->num_rates, is_11b_only_rates);
 	if (!mode->rates)
 		return -1;
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(252, mode->num_rates);
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(142, mode->num_rates);
 
@@ -1527,6 +1539,8 @@
 
 	if (hw_mode == HOSTAPD_MODE_IEEE80211G)
 		fhost_bridge_ap_start_progress(147, 0);
+	if (hw_mode == HOSTAPD_MODE_IEEE80211B)
+		fhost_bridge_ap_start_progress(253, 0);
 	return 0;
 }
 
@@ -1581,9 +1595,11 @@
 					   feat.ap_11b_only))
 			goto err;
 		mode++;
 
+		fhost_bridge_ap_start_progress(245, 0);
 		if (macsw_populate_2g_mode(mode, &feat, HOSTAPD_MODE_IEEE80211B,
 					   true))
 			goto err;
+		fhost_bridge_ap_start_progress(254, 0);
 		mode++;
 	}
 
